bootstrapping/bootstrapping.go

```diff
@@ -178,37 +230,67 @@ func start(c *cli.Context) error {
 			break
 		}
 	}
-
 	return nil
 }
 
-// Stop the bootstrapping process
-func stop(c *cli.Context) error {
-	log.Debug("cmdStop")
-
-	formatter := format.GetFormatter()
-	if err := utils.StopProcess(lockFilePath()); err != nil {
-		formatter.PrintFatal("cannot stop the bootstrapping process", err)
+func runBootstrapOnce(ctx context.Context, bootstrappingSvc *blueprint.BootstrappingService, formatter format.Formatter, thresholdLines, interval, splay int) error {
+	blueprintConfig, _, err := getBlueprintConfig(ctx, bootstrappingSvc, nil, formatter)
+	if err == nil {
+		err = applyPolicyfiles(ctx, bootstrappingSvc, blueprintConfig, formatter, thresholdLines)
 	}
-
-	log.Info("Bootstrapping routine successfully stopped")
-	return nil
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	for i := 0; err != nil && i < 3; i++ {
+		// Sleep for a configured amount of time plus a random amount of time (10 minutes plus 0 to 5 minutes, for instance)
+		ticker := time.NewTicker(time.Duration(interval+r.Intn(int(splay))) * time.Second)
+		select {
+		case <-ticker.C:
+			log.Debug("ticker")
+		case <-ctx.Done():
+			log.Debug(ctx.Err())
+			log.Debug("interrupting bootstrapping")
+			break
+		}
+		ticker.Stop()
+		blueprintConfig, _, err = getBlueprintConfig(ctx, bootstrappingSvc, nil, formatter)
+		if err == nil {
+			err = applyPolicyfiles(ctx, bootstrappingSvc, blueprintConfig, formatter, thresholdLines)
+		}
+	}
+	return err
 }
 
-// Subsidiary routine for commands processing
-func applyPolicyfiles(ctx context.Context, bootstrappingSvc *blueprint.BootstrappingService, formatter format.Formatter, thresholdLines int) error {
-	log.Debug("applyPolicyfiles")
-
+func getBlueprintConfig(ctx context.Context, bootstrappingSvc *blueprint.BootstrappingService, previousBlueprintConfig *types.BootstrappingConfiguration, formatter format.Formatter) (*types.BootstrappingConfiguration, bool, error) {
+	log.Debug("getBlueprintConfig")
 	// Inquire about desired configuration changes to be applied by querying the `GET /blueprint/configuration` endpoint. This will provide a JSON response with the desired configuration changes
-	bsConfiguration, status, err := bootstrappingSvc.GetBootstrappingConfiguration()
+	blueprintConfig, status, err := bootstrappingSvc.GetBootstrappingConfiguration()
 	if err == nil && status != 200 {
 		err = fmt.Errorf("received non-ok %d response", status)
 	}
 	if err != nil {
 		formatter.PrintError("couldn't receive bootstrapping data", err)
-		return err
+		return previousBlueprintConfig, false, err
 	}
-	err = generateWorkspaceDir()
+	updated := previousBlueprintConfig == nil
+	if !updated {
```

Comment:
Just a comment of no particular importance: "updated" is a bit ambiguous as name, as it may mean two things:
- It is not new
- Its values have not changed from last time

Reply:
What I compute in that boolean is whether there are configuration changes to apply since the last successful policyfile application...  Maybe I should change it to `newBlueprintConfigAvailable`?

Reply:
or `shouldApply`?

Reply:
well, maybe I am mistaken and it is the correct word, as you are using it in both senses...